src/app/colors/mpl-colormaps/mpl-colormap.model.ts

```diff
@@ -0,0 +1,179 @@
+import {ColorBreakpointDict} from '../color-breakpoint.model';
+import {Color} from '../color';
+import {ColorizerData} from '../colorizer-data.model';
+import {colormap_inferno_data, colormap_magma_data, colormap_plasma_data, colormap_viridis_data} from './mpl-colormaps';
+
+export type MplColormapData = Array<[number, number, number]>;
+export type MplColormapName = 'MAGMA' | 'INFERNO' | 'PLASMA' | 'VIRIDIS';
+export type ColormapStepScale = 'linear' | 'power2_steps' | 'power2_values' | 'log_steps' | 'log_values' | 'log_values_inverse' |
+    'power2_values_inverse';
+
+export interface BoundedMplColormapStepScale {
+    stepScaleName: ColormapStepScale;
+    requiresValueAbove?: number;
+    requiresValueBelow?: number;
+}
+
+export const MPL_COLORMAP_NAMES: Array<MplColormapName> = ['MAGMA', 'INFERNO', 'PLASMA', 'VIRIDIS'];
+export const COLORMAP_STEP_SCALES_WITH_BOUNDS: Array<BoundedMplColormapStepScale> = [
+    {stepScaleName: 'linear'},
+    {stepScaleName: 'power2_steps'},
+    {stepScaleName: 'power2_values', requiresValueBelow: 4000},
+    {stepScaleName: 'log_steps', requiresValueAbove: 0},
+    {stepScaleName: 'log_values', requiresValueAbove: 0},
+    {stepScaleName: 'log_values_inverse', requiresValueAbove: 0},
+    {stepScaleName: 'power2_values_inverse', requiresValueBelow: 4000}
+];
+
+export abstract class MplColormap {
+
+    static getMplColormapForName(colormapName: MplColormapName): MplColormapData {
+        switch (colormapName) {
+            case 'INFERNO':
+                return colormap_inferno_data;
+            case 'MAGMA':
+                return colormap_magma_data;
+            case 'PLASMA':
+                return colormap_plasma_data;
+            case 'VIRIDIS':
+                return colormap_viridis_data;
+        }
+    }
+
+    private static calculateStepScales(steps: number, min: number, max: number, stepScale: ColormapStepScale): Array<number> {
+        switch (stepScale) {
+            case 'linear':
+                return MplColormap.generateLinearStepFractions(steps);
+            case 'power2_steps':
+                return MplColormap.generatePowerStepFractionsSteps(steps);
+            case 'power2_values':
+                return MplColormap.generatePowStepFractionsValues(steps, min, max);
+            case 'power2_values_inverse' :
+                return MplColormap.inverseStepFractions(
+                    MplColormap.generatePowStepFractionsValues(steps, min, max)
+                );
+            case 'log_steps':
+                return MplColormap.generateLogStepFractionsSteps(steps);
+            case 'log_values':
```

Comment:
Was macht denn nun was?

Habe mal in Python etwas zum Vergleichen gecodet:
> Using colors.LogNorm normalizes the data via log10.
```Python
import matplotlib.pyplot as plt
import matplotlib.colors as colors
import numpy as np

steps = 16
start = 1
for end in [10, 100, 1_000, 10_000]:
    lognorm = colors.LogNorm(vmin=start, vmax=end)
    print('(1,', end, ')')
    for i in np.linspace(start, end, steps):
        print('{:6.2f} -> {:.5f}'.format(i, lognorm(i)))
    print()
```

```
(1, 10 )
  1.00 -> 0.00000
  1.60 -> 0.20412
  2.20 -> 0.34242
  2.80 -> 0.44716
  3.40 -> 0.53148
  4.00 -> 0.60206
  4.60 -> 0.66276
  5.20 -> 0.71600
  5.80 -> 0.76343
  6.40 -> 0.80618
  7.00 -> 0.84510
  7.60 -> 0.88081
  8.20 -> 0.91381
  8.80 -> 0.94448
  9.40 -> 0.97313
 10.00 -> 1.00000

(1, 100 )
  1.00 -> 0.00000
  7.60 -> 0.44041
 14.20 -> 0.57614
 20.80 -> 0.65903
 27.40 -> 0.71888
 34.00 -> 0.76574
 40.60 -> 0.80426
 47.20 -> 0.83697
 53.80 -> 0.86539
 60.40 -> 0.89052
 67.00 -> 0.91304
 73.60 -> 0.93344
 80.20 -> 0.95209
 86.80 -> 0.96926
 93.40 -> 0.98517
100.00 -> 1.00000

(1, 1000 )
  1.00 -> 0.00000
 67.60 -> 0.60998
134.20 -> 0.70925
200.80 -> 0.76759
267.40 -> 0.80905
334.00 -> 0.84125
400.60 -> 0.86757
467.20 -> 0.88983
533.80 -> 0.90913
600.40 -> 0.92615
667.00 -> 0.94138
733.60 -> 0.95515
800.20 -> 0.96773
866.80 -> 0.97931
933.40 -> 0.99002
1000.00 -> 1.00000

(1, 10000 )
  1.00 -> 0.00000
667.60 -> 0.70613
1334.20 -> 0.78131
2000.80 -> 0.82530
2667.40 -> 0.85652
3334.00 -> 0.88074
4000.60 -> 0.90053
4667.20 -> 0.91726
5333.80 -> 0.93176
6000.40 -> 0.94455
6667.00 -> 0.95598
7333.60 -> 0.96633
8000.20 -> 0.97578
8666.80 -> 0.98446
9333.40 -> 0.99251
10000.00 -> 1.00000
```

Reply:
Das ist ganz einfach. Die "value" Varianten verteilen über die min max Werte. Die "steps" Varianten nur über die gewählten Steps. Das sind am ende sehr unterschiedliche Funktionen. mit "steps" kann ich beliebige Wertebereiche so verteilen, dass die kleinsten/größten eine höhrere Auflösung bekommen, Mit "values" verteilt man über die echten Werte. Ich finde als Anwender beides nützlich. Würde das also auch mal evaluieren wollen.

Reply:
Stimmen die Werte denn für "values" überein?

Ich könnte mir folgendes vorstellen:
 - Linear
 - Logarithmic (das jetzige log. values)
 - Exponential (das jetzige exp. values)
 - Pattern: Decreasing Spacing (jetzige log steps)
 - Pattern: Increasing Spacing (jetzige exp. steps)

Denn nur als Entwickler weiß man, was mit steps vs. values gemeint ist.

Reply:
siehe oben